airflow/providers/google/cloud/hooks/dataprep.py

```diff
@@ -68,7 +80,7 @@ def get_jobs_for_job_group(self, job_id: int) -> dict[str, Any]:
         :param job_id: The ID of the job that will be fetched
         """
         endpoint_path = f"v4/jobGroups/{job_id}/jobs"
-        url: str = os.path.join(self._base_url, endpoint_path)
+        url: str = urljoin(self._base_url, endpoint_path)
```

Comment:
That's actually a good idea to do! We use os.path or similar instead of urljoins for urls.